src/metabase/driver/postgres.clj

```diff
@@ -279,6 +276,20 @@
   (and (str/starts-with? database-type "\"")
        (str/ends-with? database-type "\"")))
 
+(defmethod sql.qp/->honeysql [:postgres :convert-timezone]
+  [driver [_ arg to-tz from-tz]]
+  (let [clause       (sql.qp/->honeysql driver arg)
+        timestamptz? (hx/is-of-type? clause "timestamptz")]
+    (when (and timestamptz? from-tz)
+      (throw (ex-info "`timestamp with time zone` columns shouldn't have a `from timezone`" {:to-tz   to-tz
```

Comment:
what's our take on throwing during mbql->honeysql ?

Reply:
I looked in the sql query-processor and we are quite ok with throwing. And this is far preferable than emitting busted sql or getting weird errors when going from honeysql to sql. Let's throw good errors with context at the spot we can determine the error rather than having to reconstruct what went wrong in a downstream datastructure of honeysql or raw sql strings.

Reply:
it's possible the map provided to the `ex-info` needs a `:type qp.error-type/invalid-query` key value pair added to it. Not sure if the FE needs that or not.